### DISCUSSION TITLE
[ZK Stack] - <Questions about the following three features : Validium, Data Availability, Gas Token>

### DISCUSSION THREAD

**Discussion by Marius-BC-Dev** [zkSync-Community-Hub/zksync-developers#294] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.73.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

What should I look for in Validium, Data Availability, and Gas Token to be available?

### Expected Behavior

Validium, Data Availability, and Gas Token are three features I'd like to set when deploying, but I don't see the corresponding options.

### Current Behavior

Validium, Data Availability, and Gas Token are three features I'd like to set when deploying, but I don't see the corresponding options.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by zk-Lumi** [comment]
Hi @Marius-BC-Dev, Thanks for reaching out. 

Validium & Gas Token will be out soon (look for announcement on twitter/blog).

3rd party DA will also come soon, but a bit later.

If you have a project you thinking of launching with ZK Stack, please feel free to reach out to BD - bd(at)matterlabs.dev

**Reply by Marius-BC-Dev** [reply]
Thank you! Can you send me a link ?

**Reply by uF4No** [reply]
https://twitter.com/fede_intern/status/1753468927026631003

**Reply by Marius-BC-Dev** [reply]
Cool

**Reply by bxpana** [reply, marked as answer]
@Marius-BC-Dev we recently put out an update about Validiums and how this will enable DA support in the coming months https://x.com/zkSyncDevs/status/1775921157750895033

**Comment by Marius-BC-Dev** [comment]
Is there an expected point in time?

**Reply by Adinurhuda** [reply]
sorry sir please unban discord, i dont understand my discord ban.  
my id : 1098561155389194240 
user name : liger1008
I think the culprit is mods

**Reply by Marius-BC-Dev** [reply]
why？Why am I getting into discord here?

**Reply by Adinurhuda** [reply]
Sorry sir, I mean I asked to unban my discord. My previous discord was hit by a lot of bots or hacks, so I couldn't join the zksync discord anymore. I applied to unban my discord. I am active in zksync lite and zksync era transactions, and I have joined the zksync guild.

Thank You
![Screenshot 2024-02-08 155901](https://github.com/zkSync-Community-Hub/zksync-developers/assets/97208339/76db36ed-1a42-47ad-b045-3f74e1391dad)

**Comment by loriyamazaki** [comment]
To enable Validium, Data Availability, and Gas Token features during deployment, you should ensure that your deployment script or configuration file includes the necessary parameters or flags to activate these functionalities. Additionally, you may need to consult the documentation or source code of the relevant modules (such as zkEVM for Validium, Data Availability, and Gas Token) to understand how to enable these features programmatically during deployment. If these options are not available in the current version or configuration you are using, you may need to update your setup or consult with the developers of the modules for further assistance.

**Reply by Marius-BC-Dev** [reply]
Is there support for this in the code? Where can I get it?

**Comment by jodychvs** [comment]
To enable Validium, Data Availability, and Gas Token features during deployment, you typically need to ensure that the corresponding configurations are set correctly in your deployment scripts or configuration files. Ensure that you're using the appropriate versions of the related modules and that your deployment environment supports these features. If you're unable to find the necessary options, consulting the project documentation or reaching out to the project maintainers for guidance would be advisable. #Validium #DataAvailability #GasToken

**Reply by Marius-BC-Dev** [reply]
I need these features badly, who can I think of to inquire about them?

**Comment by jillalbritton** [comment]
Validium, Data Availability, and Gas Token are features that may be available in specific Layer 2 solutions or custom deployments. To enable these features, you'll typically need to configure them during the deployment process or enable them through contract parameters or settings in the corresponding protocol or system contracts.

**Reply by Marius-BC-Dev** [reply]
Give me a link or configuration instructions, would that be ok?

**Comment by syringjerry** [comment]
You can interact with smart contracts on zkSync using the Python SDK by deploying contracts, calling their functions, and querying their states. You may refer to the zkSync Python SDK documentation and explore examples provided there for guidance.

**Comment by knowenlophoroger** [comment]
The user is asking about how to set the options for Validium, Data Availability, and Gas Token when deploying, but they do not see the corresponding options. They have not provided a repository link or additional details, and have searched for similar issues but did not find any matching ones.

**Comment by Adinurhuda** [comment]
Sorry sir, can you apply to unban my discord meme, I was hit by a server band, because I was hit by a mod attack. and now I have activated it with 2FA?
username : liger1008
id : 1098561155389194240

**Reply by itsacoyote** [reply]
@Adinurhuda please email us at [support@zksync.io](mailto:support@zksync.io)
